internal/server/auth.go

```diff
@@ -7,6 +7,7 @@ import (
 	"bytes"
```

Comment:
Don't focus on this file; I just performed some changes for the unit tests (avoid keeping the connection hanging if something fails), but haven't fully tested it. I'm trying to go over the auth part once we decide the way to go.